src/Sylius/Bundle/ApiBundle/Changer/PaymentMethodChanger.php

```diff
@@ -54,6 +55,6 @@ public function changePaymentMethod(
             return $order;
         }
 
-        throw new \InvalidArgumentException('Payment method can not be set');
+        throw new PaymentMethodCannotBeSetException('Payment method can not be set');
```

Comment:
```suggestion
        throw new PaymentMethodCannotBeSetException();
```
do we need this message? The name of this exception says exactly the same thing.
BTW this class is named changer, this function is named change, so maybe this exception also should be named `PaymentMethodCannotBeChangedException` it is ofc just suggestion :)

Reply:
We can set this message in the exception's constructor as a default value for the `message` parameter. We might also consider passing more information like payment method's code. I like to create exceptions like this:
![CleanShot 2023-06-21 at 07 59 05](https://github.com/Sylius/Sylius/assets/80641364/cda82da8-8ab0-4014-9f9a-3e53c86bd198)

Also it's only a suggestion 🖖🏻